controller/src/main/java/io/pravega/controller/store/kvtable/PersistentKVTableBase.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.kvtable;
+
+import com.google.common.collect.ImmutableList;
+import io.pravega.client.tables.KeyValueTableConfiguration;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.store.Version;
+import io.pravega.controller.store.VersionedMetadata;
+import io.pravega.controller.store.kvtable.records.KVTEpochRecord;
+import io.pravega.controller.store.kvtable.records.KVTSegmentRecord;
+import io.pravega.controller.store.kvtable.records.KVTConfigurationRecord;
+import io.pravega.controller.store.kvtable.records.KVTStateRecord;
+import io.pravega.controller.store.stream.StoreException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.IntStream;
+
+
+@Slf4j
+public abstract class PersistentKVTableBase implements KeyValueTable {
```

Comment:
Please call this class AbstractTable. 
I have an issue for renaming PersistentStreamBase to AbstractStream as well.
I dont know why i created that weird name earlier. But please using a different name here. We have followed the pattern of calling the base abstract class `AbstractXXX`

Reply:
done